# Analyse App Profiles for the App Store and Google Play Markets
-----------------
**Purpose**
<div style='text-align: justify;'>
- to find mobile app profiles that are profitable for the App Store and Google Play markets.
</div>
<div style='text-align: justify;'>
- Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users. We use English for the apps we develop at our company, and we'd like to analyze only the apps that are directed toward an English-speaking audience.
</div>

**Data Description**<br>
<div style='text-align: justify;'>
    As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.
    Collecting data for over four million apps requires a significant amount of time and money, so we'll try to analyze a sample of data instead. To avoid spending resources with collecting new data ourselves, we should first try to see whether we can find any relevant existing data at no cost
</div>

In [1]:
# Library Space
import pandas as pd
import numpy as np
from csv import reader

In [2]:
# Function Space
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [3]:
# Google Play Store
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

# App Store
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In [4]:
android_header

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [5]:
ios_header

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

In [6]:
explore_data(android, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


In [7]:
explore_data(ios, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


*The Google Play data set has a dedicated discussion section, and we can see that [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) describes an error for a certain row.*

In [8]:
print(android[10472])  # incorrect row
print('\n')
print(android_header)  # header
print('\n')
print(android[0])      # correct row

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


In [9]:
print(len(android))
del android[10472]  # don't run this more than once
print(len(android))

10841
10840


**Check Duplicate App**

In [10]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

print('Total duplicate apps: ', len(duplicate_apps))
print('The duplicate apps: ')
all_duplicate_apps = pd.DataFrame(duplicate_apps, columns = ['Apps'])
all_duplicate_apps        

Total duplicate apps:  1181
The duplicate apps: 


Apps
0                          Quick PDF Scanner + OCR FREE
1                                                   Box
2                                    Google My Business
3                                   ZOOM Cloud Meetings
4                             join.me - Simple Meetings
5                                                   Box
6                                              Zenefits
7                                            Google Ads
8                                    Google My Business
9                                                 Slack
10                                   FreshBooks Classic
11                                        Insightly CRM
12          QuickBooks Accounting: Invoicing & Expenses
13                       HipChat - Chat Built for Teams
14                             Xero Accounting Software
15              MailChimp - Email, Marketing Automation
16                 Crew - Free Messaging and Scheduling
17                        Asana: organize team projects
18                                     Google Analytics
19                                      AdWords Express
20                          Accounting App - Zoho Books
21                       Invoice & Time Tracking - Zoho
22                            join.me - Simple Meetings
23    Invoice 2go — Professional Invoices and Estimates
24     SignEasy | Sign and Fill PDF and other Documents
25                         Quick PDF Scanner + OCR FREE
26                            Genius Scan - PDF Scanner
27                       Tiny Scanner - PDF Scanner App
28                         Fast Scanner : Free PDF Scan
29                     Mobile Doc Scanner (MDScan) Lite
...                                                 ...
1151      Nike Training Club - Workouts & Fitness Plans
1152                      Hangouts Dialer - Call Phones
1153                          Offline Maps & Navigation
1154              Strawberry Shortcake Ice Cream Island
1155                        Home Workout - No Equipment
1156  Home Security Camera WardenCam - reuse old phones
1157                                       Food Network
1158                            Web Browser for Android
1159          Airway Ex - Intubate. Anesthetize. Train.
1160                      FilterGrid - Cam&Photo Editor
1161        Messages, Text and Video Chat for Messenger
1162                                All Social Networks
1163                      Premier League - Official App
1164                                   Farm Heroes Saga
1165                                ESPN Fantasy Sports
1166                                    Fallout Shelter
1167                             Facebook Pages Manager
1168                               Facebook Ads Manager
1169  Who Viewed My Facebook Profile - Stalkers Visi...
1170                                      The 5th Stand
1171                                   Garena Free Fire
1172                            osmino Wi-Fi: free WiFi
1173                         Fun Kid Racing - Motocross
1174  Podcast App: Free & Offline Podcasts by Player FM
1175                                  Motorola FM Radio
1176                                 FarmersOnly Dating
1177                 Firefox Focus: The privacy browser
1178                                        FP Notebook
1179                     Slickdeals: Coupons & Shopping
1180                                               AAFP

[1181 rows x 1 columns]

In [11]:
print(android_header)
print('\n')
for i in android:
    if i[0] == 'Box':
        print(i[0:])
        print('\n')

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']




>There is no difference in all 'box' application rows

In [12]:
print(android_header)
print('\n')
for i in android:
    if i[0] == 'Instagram':
        print(i[0:])
        print('\n')

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']




> In Instagram, the higher the number of reviews, the more recent the data should be.

In [13]:
print('Expected length:', len(android)-1181)

Expected length: 9659


In [14]:
reviews_max = {}
for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and  reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews        

In [15]:
len(reviews_max)

9659

In [16]:
android_clean = []
already_added = []
for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(app)
        already_added.append(name)

In [17]:
explore_data(android_clean, 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9659
Number of columns: 13


------
<div style='text-align: justify;'>
    Remember we use English for the apps we develop at our company, and we'd like to analyze only the apps that are directed toward an English-speaking audience. However, if we explore the data long enough, we'll find that both data sets have apps with names that suggest they are not directed toward an English-speaking audience.
</div>

In [18]:
# Example of apps' name that using non English
print('Example app in Apple Store:')
print(ios[813][1])
print('\n')
print('Example app in Google :')
print(android_clean[4412][0])

Example app in Apple Store:
爱奇艺PPS -《欢乐颂2》电视剧热播


Example app in Google :
中国語 AQリスニング


------
<div style='text-align: justify;'>
The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the ASCII (American Standard Code for Information Interchange) system. Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not. If the number is equal to or less than 127, then the character belongs to the set of common English characters.
</div>

In [19]:
def is_english(string):
    for char in string:
        if ord(char) > 127:
            return False
    return True
print('Instagram is probably English Name (',is_english('Instagram'),')')
print('\n')
print('爱奇艺PPS -《欢乐颂2》电视剧热播 is probably English Name (',is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'),')')
print('\n')
print('Docs To Go™ Free Office Suite is probably English Name (',is_english('Docs To Go™ Free Office Suite'),')')
print('\n')
print('Instachat 😜 is probably English Name (',is_english('Instachat 😜'),')')
print('\n')

Instagram is probably English Name ( True )


爱奇艺PPS -《欢乐颂2》电视剧热播 is probably English Name ( False )


Docs To Go™ Free Office Suite is probably English Name ( False )


Instachat 😜 is probably English Name ( False )




------
<div style='text-align: justify;'>
If we are using this function, we will lose useful data since there are many English apps will be incorrectly labeled as non-English. Example: Instachat and Docs To Go. 
</div>

***The reason is emojis fall outside the ASCII range (over 127) ***

In [20]:
# To prevent that, we change our function
def is_english(string):
    non_ascii = 0
    for char in string:
        if ord(char) > 127:
            non_ascii += 1
    if non_ascii > 3:
        return False
    else:
        return True

In [21]:
print('爱奇艺PPS -《欢乐颂2》电视剧热播 is probably English Name (',is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'),')')
print('\n')
print('Docs To Go™ Free Office Suite is probably English Name (',is_english('Docs To Go™ Free Office Suite'),')')
print('\n')
print('Instachat 😜 is probably English Name (',is_english('Instachat 😜'),')')
print('\n')

爱奇艺PPS -《欢乐颂2》电视剧热播 is probably English Name ( False )


Docs To Go™ Free Office Suite is probably English Name ( True )


Instachat 😜 is probably English Name ( True )




In [22]:
english_android = []
english_ios = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        english_android.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        english_ios.append(app)
        
explore_data(english_android, 0, 3, True)
print('\n')
explore_data(english_ios, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

In [23]:
free_english_android = []
free_english_ios = []

for app in english_android:
    price = app[7]
    if price == '0':
        free_english_android.append(app)
        

for app in english_ios:
    price = app[4]
    if price == '0.0':
        free_english_ios.append(app)
        
print(len(free_english_android))
print(len(free_english_ios))

8864
3222


<div style = 'text-align: justify'>
As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.<br><br>

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:<br><br>

1. Build a minimal Android version of the app, and add it to Google Play.<br>
2. If the app has a good response from users, we then develop it further.<br>
3. If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.<br><br>

Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.<br><br>

Let's begin the analysis by getting a sense of the most common genres for each market. For this, we'll build a frequency table for the prime_genre column of the App Store data set, and the Genres and Category columns of the Google Play data set.
</div>

-------
<div style='text-align: justify'>
Our conclusion was that we'll need to build a frequency table for the prime_genre column of the App Store data set, and for the Genres and Category columns of the Google Play data set.<br><br>

We'll build two functions we can use to analyze the frequency tables: <br> <br>

1. One function to generate frequency tables that show percentages <br>
2. Another function we can use to display the percentages in a descending order
</div>

In [24]:
def freq_table(dataset, index):
    table = {}
    total = 0
    table_percentage = {}
    
    for app in dataset:
        total += 1
        value = app[index]
        
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentage[key] = percentage
    
    return table_percentage

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
    
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], '=', entry[0])
    

In [25]:
display_table(free_english_ios, -5)

Games = 58.16263190564867
Entertainment = 7.883302296710118
Photo & Video = 4.9658597144630665
Education = 3.662321539416512
Social Networking = 3.2898820608317814
Shopping = 2.60707635009311
Utilities = 2.5139664804469275
Sports = 2.1415270018621975
Music = 2.0484171322160147
Health & Fitness = 2.0173805090006205
Productivity = 1.7380509000620732
Lifestyle = 1.5828677839851024
News = 1.3345747982619491
Travel = 1.2414649286157666
Finance = 1.1173184357541899
Weather = 0.8690254500310366
Food & Drink = 0.8069522036002483
Reference = 0.5586592178770949
Business = 0.5276225946617008
Book = 0.4345127250155183
Navigation = 0.186219739292365
Medical = 0.186219739292365
Catalogs = 0.12414649286157665


-----
<div style = 'text-align:justify'>
We can see that among the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.<br><br>

The general impression is that App Store (at least the part containing free English apps) ___***is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare.***___
</div>

In [26]:
display_table(free_english_android, 1)

FAMILY = 18.907942238267147
GAME = 9.724729241877256
TOOLS = 8.461191335740072
BUSINESS = 4.591606498194946
LIFESTYLE = 3.9034296028880866
PRODUCTIVITY = 3.892148014440433
FINANCE = 3.7003610108303246
MEDICAL = 3.531137184115524
SPORTS = 3.395758122743682
PERSONALIZATION = 3.3167870036101084
COMMUNICATION = 3.2378158844765346
HEALTH_AND_FITNESS = 3.0798736462093865
PHOTOGRAPHY = 2.944494584837545
NEWS_AND_MAGAZINES = 2.7978339350180503
SOCIAL = 2.6624548736462095
TRAVEL_AND_LOCAL = 2.33528880866426
SHOPPING = 2.2450361010830324
BOOKS_AND_REFERENCE = 2.1435018050541514
DATING = 1.861462093862816
VIDEO_PLAYERS = 1.7937725631768955
MAPS_AND_NAVIGATION = 1.3989169675090252
FOOD_AND_DRINK = 1.2409747292418771
EDUCATION = 1.1620036101083033
ENTERTAINMENT = 0.9589350180505415
LIBRARIES_AND_DEMO = 0.9363718411552346
AUTO_AND_VEHICLES = 0.9250902527075812
HOUSE_AND_HOME = 0.8235559566787004
WEATHER = 0.8009927797833934
EVENTS = 0.7107400722021661
PARENTING = 0.6543321299638989
ART_AND_DESIGN = 

-----
<div style = 'text-align:justify'>
The landscape seems significantly different on Google Play: there are not that many apps designed for fun, and ***it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.).*** However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids. <br> <br>

Even so, practical apps seem to have a better representation on Google Play compared to App Store.
</div>

In [27]:
display_table(free_english_android, -4)

Tools = 8.449909747292418
Entertainment = 6.069494584837545
Education = 5.347472924187725
Business = 4.591606498194946
Productivity = 3.892148014440433
Lifestyle = 3.892148014440433
Finance = 3.7003610108303246
Medical = 3.531137184115524
Sports = 3.463447653429603
Personalization = 3.3167870036101084
Communication = 3.2378158844765346
Action = 3.1024368231046933
Health & Fitness = 3.0798736462093865
Photography = 2.944494584837545
News & Magazines = 2.7978339350180503
Social = 2.6624548736462095
Travel & Local = 2.3240072202166067
Shopping = 2.2450361010830324
Books & Reference = 2.1435018050541514
Simulation = 2.0419675090252705
Dating = 1.861462093862816
Arcade = 1.8501805054151623
Video Players & Editors = 1.7712093862815883
Casual = 1.7599277978339352
Maps & Navigation = 1.3989169675090252
Food & Drink = 1.2409747292418771
Puzzle = 1.128158844765343
Racing = 0.9927797833935018
Role Playing = 0.9363718411552346
Libraries & Demo = 0.9363718411552346
Auto & Vehicles = 0.9250902527075

-----
<div style = 'text-align:justify'>
The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward. <br> <br>

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps.
</div>

** Most Popular Apps by Genre on the App Store **

In [28]:
genres_ios = freq_table(free_english_ios, -5)
table = []
for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in free_english_ios:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    table.append((avg_n_ratings, genre))
sorted_table = sorted(table, reverse = True)
for i in sorted_table:
    print(i[1], '=', i[0])

Navigation = 86090.33333333333
Reference = 74942.11111111111
Social Networking = 71548.34905660378
Music = 57326.530303030304
Weather = 52279.892857142855
Book = 39758.5
Food & Drink = 33333.92307692308
Finance = 31467.944444444445
Photo & Video = 28441.54375
Travel = 28243.8
Shopping = 26919.690476190477
Health & Fitness = 23298.015384615384
Sports = 23008.898550724636
Games = 22788.6696905016
News = 21248.023255813954
Productivity = 21028.410714285714
Utilities = 18684.456790123455
Lifestyle = 16485.764705882353
Entertainment = 14029.830708661417
Business = 7491.117647058823
Education = 7003.983050847458
Catalogs = 4004.0
Medical = 612.0


-----
<div style = 'text-align:justify'>
On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together.

</div>

In [29]:
for i in free_english_ios:
    if i[-5] == 'Navigation':
        print(i[1], '=', i[5])

Waze - GPS Navigation, Maps & Real-time Traffic = 345046
Google Maps - Navigation & Transit = 154911
Geocaching® = 12811
CoPilot GPS – Car Navigation & Offline Maps = 3582
ImmobilienScout24: Real Estate Search in Germany = 187
Railway Route Search = 5


-----
<div style = 'text-align:justify'>
The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number. <br> <br>

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. ***The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold.***
</div>

In [30]:
for i in free_english_ios:
    if i[-5] == 'Reference':
        print(i[1], '=', i[5])

Bible = 985920
Dictionary.com Dictionary & Thesaurus = 200047
Dictionary.com Dictionary & Thesaurus for iPad = 54175
Google Translate = 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran = 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition = 17588
Merriam-Webster Dictionary = 16849
Night Sky = 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) = 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools = 4693
GUNS MODS for Minecraft PC Edition - Mods Tools = 1497
Guides for Pokémon GO - Pokemon GO News and Cheats = 826
WWDC = 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free = 718
VPN Express = 14
Real Bike Traffic Rider Virtual Reality Glasses = 8
教えて!goo = 0
Jishokun-Japanese English Dictionary & Translator = 0


-----
<div style = 'text-align:justify'>
However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app. <br> <br>

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.
</div>

**Most Popular Apps by Genre on Google Play**

In [32]:
categories_android = freq_table(free_english_android, 1)
table = []
for category in categories_android:
    total = 0
    len_category = 0
    for app in free_english_android:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    table.append((avg_n_installs, category))
sorted_table = sorted(table, reverse = True)
for i in sorted_table:
    print(i[1], '=', i[0])

COMMUNICATION = 38456119.167247385
VIDEO_PLAYERS = 24727872.452830188
SOCIAL = 23253652.127118643
PHOTOGRAPHY = 17840110.40229885
PRODUCTIVITY = 16787331.344927534
GAME = 15588015.603248259
TRAVEL_AND_LOCAL = 13984077.710144928
ENTERTAINMENT = 11640705.88235294
TOOLS = 10801391.298666667
NEWS_AND_MAGAZINES = 9549178.467741935
BOOKS_AND_REFERENCE = 8767811.894736841
SHOPPING = 7036877.311557789
PERSONALIZATION = 5201482.6122448975
WEATHER = 5074486.197183099
HEALTH_AND_FITNESS = 4188821.9853479853
MAPS_AND_NAVIGATION = 4056941.7741935486
FAMILY = 3695641.8198090694
SPORTS = 3638640.1428571427
ART_AND_DESIGN = 1986335.0877192982
FOOD_AND_DRINK = 1924897.7363636363
EDUCATION = 1833495.145631068
BUSINESS = 1712290.1474201474
LIFESTYLE = 1437816.2687861272
FINANCE = 1387692.475609756
HOUSE_AND_HOME = 1331540.5616438356
DATING = 854028.8303030303
COMICS = 817657.2727272727
AUTO_AND_VEHICLES = 647317.8170731707
LIBRARIES_AND_DEMO = 638503.734939759
PARENTING = 542603.6206896552
BEAUTY = 51315

<div style='text-align:justify'>
On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:
</div>

In [33]:
for app in free_english_android:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

<div style='text-align:justify'>
If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times. We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.). <br> <br>

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.
</div>

In [34]:
for app in free_english_android:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

<div style = 'text-align:justify'>
The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:
</div>

## Conclusion
-------
<div style = 'text-align:justify'>
In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets. <br> <br>

***I conclude that book and knowledge niche for our app could be profitable for both Google Play and App Store market.*** The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.
</div>